steel/tests/steel.rs

```diff
@@ -0,0 +1,236 @@
+        }
+
+        /// Tests accessing the code of the EOA account 0x0000000000000000000000000000000000000000.
+        function testEoaAccount() external view returns (uint256 size) {
+            assembly { size := extcodesize(0) }
+        }
+
+        /// Tests the blockhash opcode.
+        function testBlockhash() external view returns (bytes32 h) {
+            assembly { h := blockhash(sub(number(), 256)) }
+        }
+
+        /// Tests retrieving the chain ID.
+        function testChainid() external view returns (uint256) {
+            return block.chainid;
+        }
+
+        /// Tests retrieving the address of the sender of the transaction.
+        function testOrigin() external view returns (address) {
+            return tx.origin;
+        }
+
+        /// Tests retrieving the gas price.
+        function testGasprice() external view returns (uint256) {
+            return tx.gasprice;
+        }
+
+        /// Tests calling multiple contracts with the same and different storage.
+        function testMuliContractCalls() external view returns (uint256) {
+            return VALUE0.value() + VALUE42A.value() + VALUE42B.value();
+        }
+    }
+);
+
+type TestProvider = FillProvider<
+    JoinFill<RecommendedFiller, WalletFiller<EthereumWallet>>,
+    AnvilProvider<RootProvider<Http<Client>>, Http<Client>>,
+    Http<Client>,
+    Ethereum,
+>;
+
+/// Returns an Anvil provider with the deployed [SteelTest] contract.
+async fn test_provider() -> TestProvider {
+    let provider = ProviderBuilder::new()
+        .with_recommended_fillers()
+        .on_anvil_with_wallet_and_config(|anvil| anvil.args(["--hardfork", "cancun"]));
+    let node_info = provider.anvil_node_info().await.unwrap();
+    log::info!("Anvil started: {:?}", node_info);
+    let instance = SteelTest::deploy(&provider).await.unwrap();
+    assert_eq!(*instance.address(), STEEL_TEST_CONTRACT);
+
+    provider
+}
+
+#[test(tokio::test)]
+async fn precompile() {
+    let result = common::eth_call(
+        test_provider().await,
+        STEEL_TEST_CONTRACT,
+        SteelTest::testPrecompileCall {},
+        CallOptions::new(),
+    )
+    .await;
+    assert_eq!(
+        result._0,
+        b256!("e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855")
+    );
+}
+
+#[test(tokio::test)]
+async fn nonexistent_account() {
+    let result = common::eth_call(
+        test_provider().await,
+        STEEL_TEST_CONTRACT,
+        SteelTest::testNonexistentAccountCall {},
+        CallOptions::new(),
+    )
+    .await;
+    assert_eq!(result.size, uint!(0_U256));
+}
+
+#[test(tokio::test)]
+async fn eoa_account() {
+    let result = common::eth_call(
+        test_provider().await,
+        STEEL_TEST_CONTRACT,
+        SteelTest::testEoaAccountCall {},
+        CallOptions::new(),
+    )
+    .await;
+    assert_eq!(result.size, uint!(0_U256));
+}
+
+#[tokio::test] // Anvil crashes, if logging is enabled
```

Comment:
If this is something we can address, can you create a quick issue it and add it's number here?

Reply:
Seems like this was somehow fixed in the latest version, so now even logging works here.